ambry-cloud/src/main/java/com/github/ambry/cloud/CloudStorageCompactor.java

```diff
@@ -103,47 +106,80 @@ public int compactPartitions() {
       logger.info("Skipping compaction as no partitions are assigned.");
       return 0;
     }
+    List<PartitionId> partitionSnapshot = new ArrayList<>(partitions);
+    long compactionStartTime = System.currentTimeMillis();
+    long timeToQuit = System.currentTimeMillis() + compactionTimeLimitMs;
+    int compactionInProgress = 0;
+    doneLatch.set(new CountDownLatch(1));
+    int totalBlobsPurged = 0;
+    int compactedPartitionCount = 0;
+    try {
+      while (true) {
```

Comment:
This could be: while (!partitionSnapshot.isEmpty())

Reply:
yes.. will fix this.